soh/soh/Enhancements/randomizer/3drando/logic.cpp

```diff
@@ -785,7 +785,8 @@ namespace Logic {
                            (Bridge.Is(RAINBOWBRIDGE_REWARDS)    && StoneCount + MedallionCount >= BridgeRewardCount.Value<uint8_t>())  ||
                            (Bridge.Is(RAINBOWBRIDGE_DUNGEONS)   && DungeonCount >= BridgeDungeonCount.Value<uint8_t>())                ||
                            (Bridge.Is(RAINBOWBRIDGE_TOKENS)     && GoldSkulltulaTokens >= BridgeTokenCount.Value<uint8_t>()) ||
-                           (Bridge.Is(RAINBOWBRIDGE_GREG)       && Greg);
+                           (Bridge.Is(RAINBOWBRIDGE_GREG)       && Greg) ||
+                           (Bridge.Is(RAINBOWBRIDGE_WILD_GREG) && StoneCount + MedallionCount >= BridgeRewardCount.Value<uint8_t>());
```

Comment:
Is the intention here to have logic not take finding Greg into account? With this logic a seed will not be generated with the expectation of finding Greg as one of the rewards needed to open the bridge

Reply:
Yes. The intention is for him to be a possible reward, not a guaranteed one. Meaning you could do a whole seed logically and never find him, or you can find him and skip some of the logic. The idea is to make him more of an exciting item to find, rather than a required item. That's what makes it a wildcard and not just adding him as a dungeon reward.

Reply:
i think both ways could be fun, so instead of a conditional checkbox, we could have a conditional dropdown when choosing a bridge type greg can be a wildcard for

* Standard Rewards
  * Greg does not change logic, Greg does not help open the bridge, max number of rewards on slider does not change
* Greg as Wildcard
  * Greg does not change logic, Greg helps open the bridge, max number of rewards on slider does not change
* Greg as Reward
  * Greg does change logic (can be part of expected path for opening bridge), Greg helps open bridge, max number of rewards on slider increases by 1 to account for Greg
  
If you need any help with implementation feel free to reach out. I'm not sure what my availability will be over the next few days but when I have some time I'd be happy to hop in and PR to your branch if you'd like

Reply:
I've been trying to just add it as a checkbox for a while now and can't seem to figure it out, so I definitely will need some help haha

Reply:
I love all these ideas though! I'm just not sure if I know enough to be able to do this all myself.

Reply:
so if you look at the bridge switch case here https://github.com/HarbourMasters/Shipwright/blob/061e2326857920344f952595a48e3d3547efe1a9/soh/soh/Enhancements/randomizer/randomizer.cpp#L3325-L3359 you could use that as an example of conditionally showing something based on bridge setting

one way to handle this would be to just add a dropdown (`EnhancementCombobox`) above the `UIWidgets::EnhancementSliderInt` in each of the cases in that switch

for the combobox options, you could follow the pattern of an existing one - see:
https://github.com/HarbourMasters/Shipwright/blob/061e2326857920344f952595a48e3d3547efe1a9/soh/soh/Enhancements/randomizer/randomizer.cpp#L3255
 https://github.com/HarbourMasters/Shipwright/blob/061e2326857920344f952595a48e3d3547efe1a9/soh/soh/Enhancements/randomizer/randomizer.cpp#L3076
 
this would probably mean a bit of copypasta, but that can be cleaned up after the functionality is nailed down

Reply:
The UI part I can do, it's the additional functionality I might have trouble with. I'm still trying some stuff out though.